main.tf

```diff
@@ -195,6 +195,13 @@ resource "google_pubsub_subscription" "push_subscriptions" {
         audience              = lookup(each.value, "audience", "")
       }
     }
+
+    dynamic "no_wrapper" {
+      for_each = lookup(each.value, "no_wrapper", "") ? [true] : []
```

Comment:
Rather than a bool, would it make more sense to use a nested block `no_wrapper  = {write_metadata: false}` with a default of `null`?  That would more closely follow the [provider](https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/pubsub_subscription.html#nested_no_wrapper) and allow for easy addition of additional keys in the future.